forc-pkg/src/pkg.rs

```diff
@@ -279,37 +271,54 @@ impl BuildPlan {
             graph,
             path_map,
             compilation_order,
-            removed_deps: vec![],
         })
     }
 
     /// Attempt to load the build plan from the `Lock`.
-    pub fn from_lock(proj_path: &Path, lock: &Lock, sway_git_tag: &str) -> Result<Self> {
+    pub fn from_lock(
+        proj_path: &Path,
+        lock: &Lock,
+        sway_git_tag: &str,
+    ) -> Result<(Self, Vec<String>)> {
```

Comment:
`Vec<String>` is a very ambiguous return type and it's not at all clear from reading the function signature what it represents. Either we should update the docs for this function, or use a type to represent the deps removed due to invalid paths, or both.